docker-compose.yml

```diff
@@ -1,35 +1,120 @@
-version: "3"
+version: "3.4"
+
+x-healthcheck: &healthcheck-interval
+  interval: 1s
+  timeout: 1s
+  start_period: 10s
+  # TODO: https://github.com/docker/compose/issues/10461
+  # interval: 10s
+  # start_interval: 1s
+
+x-environment: &environment
+    RUST_BACKTRACE: true
+
+    DOCSRS_PREFIX: /opt/docsrs/prefix
+
+    DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@db
```

Comment:
not taking this from `.env` would make it harder to mix containerized services with running the services directly (or the CLI). 

Is this intentional?

Reply:
Inside containers it needs to be `@db`, outside containers it needs to be `@localhost:15432`, so we can't have a shared `.env` for it anyway. That also somewhat answers the above question, you can't have the containers loading `.env` because that will override the pre-configured variables to use ones for local dev instead of ones for in-container dev.